articles/how-to-setup-ci-cd-to-vue-with-firebase-hosting-and-github-actions/index.md

```diff
@@ -0,0 +1,223 @@
+
+Well, the solution to this is to automate the entire deployment process. And, this can be done by applying CI/CD to our project. And here, we are going to do that using Vue, Firebase Hosting, and GitHub workflows. The overall process automates the process of software development and management and makes it easy to test and deploy the final results. The bottom line is to automatically build and deploy the updates to the main server after each pull request or change has been merged to the master branch of our GitHub repository. Hence, the overall process involves:
+
+- Setting up the Vue project.
+- Pushing the Vue project to the GitHub repository.
+- Setting up the Firebase project.
+- Setting up Firebase hosting in Vue project using latest firebase-tools.
+- Configuring the connection between Firebase Hosting and GitHub by applying GitHub workflows.
+
+What will happen as a result:
+
+- Push the code to the master branch or create a pull request to merge the branch.
+- Triggers the GitHub actions workflow scripts.
+- After the successful merge action, the changes will be automatically deployed to the live website without having to do any manual configurations.
+- We can also include the testing scripts in the workflow scripts to automate testing before merging.
+
+### Pre-requisites
+
+- [Git Version Control](https://github.com/git-for-windows/git/releases/download/v2.29.1.windows.1/Git-2.29.1-64-bit.exe)
+- [GitHub Account](http://github.com/)
+- [Firebase Account](https://firebase.google.com/)
+- [Vue CLI](https://cli.vuejs.org/)
+- Knowledge of [Vue JavaScript framework](https://vuejs.org/).
+
+### Step 1: Vue Boilerplate
+
+The first step is to get out Vue project ready. For the Vue boilerplate project, we are going to make use of Vue CLI. So first, we need to install the latest version of Vue CLI globally in our system. It allows us to download the starter Vue template based on various configurations. The required packages are pre-installed. It also enables us to manage the Vue project. To install it, we need to run the following command in our system terminal:
+
+```bash
+npm install @vue/cli -g
+```
+
+After Vue CLI has been installed, we can now start a new Vue Project. For that, we need to run the following command in the directory where we want to download the project:
+
+```bash
+vue create vue_cicd
+```
+
+After running the command instead of executing the automatic setup of features, we need to select manual configuration. In the manual configuration option, we will need to specify which Vue features to include. The required features are marked as shown in the screenshot below:
+
+![create new vue project with cli](1-create-new-vue-project-with-cli.png)
```

Comment:
add correct links to the images or they will not show in production

Reply:
where this link that your check? I check on [this](https://github.com/section-engineering-education/engineering-education/blob/bc20c3b9ad6d798af223146805fea576e8e3d48a/articles/how-to-setup-ci-cd-to-vue-with-firebase-hosting-and-github-actions/index.md) it also works fine
also in the [previous post](https://github.com/section-engineering-education/engineering-education/pull/920/files#diff-0457992e9101c3658a0b51f5ca278720f3bf41832126ba349dbfe25eec3e72f4)  I already do this way and also pass review

Reply:
That's interesting.
If that is the case then it's okay. 
However, I highly recommend you make the changes to the image links and make use of the standard system. The reason for this is because the publishing happens via ci/cd and relative linking might not work right all the time. To be safe make the change to the image links so we proceed to the final review.

Reply:
@Worawat101 
The correct image syntax would be /engineering-education/ci-cd-setup-vue/(image-file-name)

It was corrected during my final review perhaps. 

Please be sure to add the images with the proper syntax 👍 
Thank you

cc @katungi

Reply:
Thanks @hectorkambow for the clarification.
cheers